### PR TITLE
Add Recase typings

### PR DESCRIPTION
**Pull request by 18steps**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/relaxed-json/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+export function parse(text: string, reviver: (this: {}, key: string, value: any) => any): {};
+export function parse(text: string, opts?: {
+    relaxed?: boolean,
+    warnings?: boolean,
+    tolerant?: boolean,
+    duplicate?: boolean,
+}): {};
+export function stringify(obj: any): string;
+
+export default {
```
Comment (RyanCavanaugh) [inline types/relaxed-json/index.d.ts]: This object is not the default export of the package!

File: types/relaxed-json/relaxed-json-tests.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import RJSON from 'relaxed-json';
```
Comment (RyanCavanaugh) [inline types/relaxed-json/relaxed-json-tests.ts]: This is not how you would use the package in practice.

File: types/recase/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// Type definitions for Recase v1.0.4
+// Project: https://www.npmjs.com/package/recase
+// Definitions by: Mikal Madsen <mikal.madsen@gmail.com>
```
Comment (andy-ms) [inline types/recase/index.d.ts]: Prefer your github page like you used for relaxed-json

File: types/relaxed-json/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+// Definitions by: Mikal Madsen <https://github.com/18steps>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export function transform(text: string): string;
+export function parse(text: string, reviver: (this: {}, key: string, value: any) => any): {};
+export function parse(text: string, opts?: {
+    relaxed?: boolean,
+    warnings?: boolean,
+    tolerant?: boolean,
+    duplicate?: boolean,
```
Comment (andy-ms) [inline types/relaxed-json/index.d.ts]: One of the options may be a reviver function too. Also, `tolerant` doesn't appear in the documentation.

File: types/relaxed-json/index.d.ts (hunk 1 above)
Comment (andy-ms) [inline types/relaxed-json/index.d.ts]: Yes, this line should simply be removed. The package doesn't have a default export, it has several named exports.

File: types/relaxed-json/tslint.json [hunk 5]
```diff
@@ -0,0 +1 @@
+{ "extends": "../tslint.json" }
```
Comment (andy-ms) [inline types/relaxed-json/tslint.json]: Change these to `"extends": "dtslint/dt.json"`.

Comment (18steps) [inline types/relaxed-json/tslint.json]: > Failed to load ...\DefinitelyTyped\types\relaxed-json\tslint.json: Invalid "extends" configuration value - could not require "dtslint/dt.json". Review the Node lookup algorithm (https://nodejs.org/api/modules.html#modules_all_together) for the approximate method TSLint uses to find the referenced configuration file.

The original value is what's generated by the generator script.

Comment (andy-ms) [inline types/relaxed-json/tslint.json]: The generator script is out of date (Microsoft/dts-gen#50), and you may be running an older version of dtslint if you're getting the above error.

**Comment by 18steps** [comment]
For relaxed-json I'm exporting a Reviver type, but this is based on [native JSON.parse](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse) so I'm wondering if there's a native type I should import, and if so where I can find it.

**Comment by andy-ms** [comment]
The builtin `JSON.parse` just uses `reviver?: (key: any, value: any) => any`. (You can use goto-definition to see TypeScript's builtin type declarations.) I think yours is better though, see Microsoft/TypeScript#6955.

**Pull request merged**